server/catgenome/src/main/java/com/epam/catgenome/manager/reference/MotifSearchManager.java

```diff
@@ -155,6 +163,22 @@ private MotifSearchResult searchRegionMotifs(final MotifSearchRequest request, f
                                 && motif.getStart() <= request.getEndPosition())
                         .collect(Collectors.toList());
 
+        final List<GeneFile> geneFiles = reference.getGeneFile() != null ?
+                Collections.singletonList(reference.getGeneFile()) : Collections.emptyList();
+        String geneIdsString;
+        String geneNamesString;
+        Set<GeneInfo> geneIds;
+
+        final NggbIntervalTreeMap<List<Gene>> intervalMap = featureIndexManager.loadGenesIntervalMap(geneFiles,
```

Comment:
To prevent unnecessary search calls, lets check that `searchResult ` is not empty